app_utils/gpio.py

```
@@ -1402,42 +1420,93 @@ def _stop_flash(self, pin: int) -> None:
+                log_entry.success = event.success
+                log_entry.error_message = event.error_message
 
-            self.db_session.add(log_entry)
-            self.db_session.commit()
+                self.db_session.commit()
+                event.record_id = log_entry.id
 
             if self.logger:
                 self.logger.debug(f"Saved GPIO activation log for pin {event.pin}")
 
         except Exception as exc:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)